File: app/src/main/java/com/rendersoncs/reportform/adapter/ReportResumeAdapter.java
package com.rendersoncs.reportform.adapter;

import android.content.Context;
import android.os.Bundle;
import android.util.Base64;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.rendersoncs.reportform.R;
import com.rendersoncs.reportform.fragment.FullPhotoFragment;
import com.rendersoncs.reportform.itens.ReportResumeItems;
import com.rendersoncs.reportform.listener.OnItemListenerClicked;

import java.util.Arrays;
import java.util.List;

public class ReportResumeAdapter extends RecyclerView.Adapter<ReportResumeAdapter.ViewHolder> {

    private List<ReportResumeItems> repoResumeList;
    private Context context;

    private OnItemListenerClicked onItemListenerClicked;

    public void setOnItemListenerClicked(OnItemListenerClicked onItemListenerClicked) {
        this.onItemListenerClicked = onItemListenerClicked;
    }

    public ReportResumeAdapter(List<ReportResumeItems> repoResumeList, Context context){
        this.repoResumeList = repoResumeList;
        this.context = context;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int i) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.activity_report_resume_list, parent, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, final int i) {
        int position = holder.getAdapterPosition();
        final ReportResumeItems resumeItems = repoResumeList.get(position);

        holder.resume_title.setText(resumeItems.getTitle());
        holder.resume_description.setText(resumeItems.getDescription());
        holder.resume_conformity.setText(resumeItems.getConformity());
        holder.resume_note.setText(resumeItems.getNote());

        holder.resume_photo.setOnClickListener(view -> {
            onItemListenerClicked.fullPhoto(position);
        });
    }

    @Override
    public int getItemCount() {
        return this.repoResumeList.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder  {

        private TextView resume_title, resume_description, resume_conformity, resume_note, resume_photo;

        private ViewHolder(@NonNull View itemView) {
            super(itemView);

            resume_title = itemView.findViewById(R.id.jo_title);
            resume_description = itemView.findViewById(R.id.jo_description);
            resume_conformity = itemView.findViewById(R.id.jo_conformity);
            resume_note = itemView.findViewById(R.id.jo_note);
            resume_photo = itemView.findViewById(R.id.jo_photo);
        }
    }
}

File: app/src/main/java/com/rendersoncs/report/view/resume/ReportResumeAdapter.kt
package com.rendersoncs.report.view.resume

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.AsyncListDiffer
import com.rendersoncs.report.infrastructure.util.ItemCallBack
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.rendersoncs.report.R
import com.rendersoncs.report.databinding.ItemReportResumeListBinding
import com.rendersoncs.report.infrastructure.constants.ReportConstants
import com.rendersoncs.report.model.ReportResumeItems

class ReportResumeAdapter(private val onClickItem: (ReportResumeItems) -> Unit) :
        RecyclerView.Adapter<ReportResumeAdapter.ViewHolder>() {

    val differ = AsyncListDiffer(this, ItemCallBack<ReportResumeItems>())

    override fun onCreateViewHolder(parent: ViewGroup, i: Int): ViewHolder {
        val binding = ItemReportResumeListBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, i: Int) {
        val position = holder.layoutPosition
        val resumeItems = differ.currentList[position]

        holder.resumeTitle.text = resumeItems.title
        holder.resumeDescription.text = resumeItems.description
        holder.resumeConformity.text = when (resumeItems.conformity) {
            1 -> { holder.resumeConformity.context.getString(R.string.according) }
            2 -> { holder.resumeConformity.context.getString(R.string.not_applicable) }
            else -> holder.resumeConformity.context.getString(R.string.not_according)
        }
        holder.resumeNote.text = resumeItems.note.ifEmpty {
            holder.resumeNote.context.getString(R.string.label_not_observation)
        }

        holder.resumePhoto.setOnClickListener {
            onClickItem.invoke(resumeItems)
        }

        this.decorationItems(holder, resumeItems)
    }

    private fun decorationItems(holder: ViewHolder, resumeItems: ReportResumeItems) {
        val photo = resumeItems.photo

        if (photo.isEmpty() || photo == ReportConstants.PHOTO.NOT_PHOTO) {
            holder.resumePhoto.imageAlpha = R.drawable.ic_broken_image
        } else {
            Glide.with(holder.resumePhoto.context).load(photo).centerCrop().into(holder.resumePhoto)
        }

        val according = holder.itemView.context.resources.getString(R.string.according)
        val notApplicable = holder.itemView.context.resources.getString(R.string.not_applicable)

        when (holder.resumeConformity.text) {
            according -> {
                holder.resumeConformity.apply {
                    setTextColor(
                            ContextCompat.getColor(holder.resumeConformity.context,
                                    R.color.colorRadioC))
                }
            }
            notApplicable -> {
                holder.resumeConformity.apply {
                    setTextColor(
                            ContextCompat.getColor(holder.resumeConformity.context,
                                    R.color.colorRadioNA))
                }
            }
            else -> {
                holder.resumeConformity.apply {
                    setTextColor(
                            ContextCompat.getColor(holder.resumeConformity.context,
                                    R.color.colorRadioNC))
                }
            }
        }
    }

    override fun getItemCount(): Int {
        return differ.currentList.size
    }

    class ViewHolder(val binding: ItemReportResumeListBinding) : RecyclerView.ViewHolder(binding.root) {
        val resumeTitle = binding.joTitle
        val resumeDescription = binding.joDescription
        val resumeConformity = binding.joConformity
        val resumeNote = binding.joNote
        val resumePhoto = binding.joPhoto
    }
}